lib/watch/fork.js

```diff
@@ -0,0 +1,35 @@
+const chalk = require('chalk')
+const { stop, runFastify } = require('../../start')
+
+const GRACEFUL_SHUT = 'GRACEFUL SHUTDOWN'
+const READY = 'ready'
+
+const fastify = runFastify(process.argv.splice(2))
+const type = process.env.childEvent
+
+process.send({ type: type, err: null })
+
+fastify.ready(err => {
+  process.send({ type: READY, err: err })
+  if (err) { stop(err) }
+})
+
+process.on('message', function (event) {
+  if (event === GRACEFUL_SHUT) {
+    const timeout = setTimeout(function () {
+      process.exit(1)
+    }, 5000)
+
+    fastify.close(() => {
+      timeout.unref()
+      process.exit(0)
+    })
+  }
+})
+
+process.on('uncaughtException', (err) => {
+  fastify.close()
+  console.log(err)
+  console.log(chalk.red('[fastify-cli] app crashed - waiting for file changes before starting...'))
+  process.exit(1)
```

Comment:
we should have:

```js
process.on('uncaughtException', (err) => {
  console.log(err)
  fastify.close(exit) 
  setTimeout(exit, 5000).unref() // so that we wait only 5s for a graceful exit.

function exit () {
    console.log(chalk.red('[fastify-cli] app crashed - waiting for file changes before starting...'))
      process.exit(1)
  })
 
})
```

Reply:
`fastify.close(exit)` it's looks that only happend fastify normally startup. so we should detect fastify is normally runing, it's normally running that graceful exit happen, other case I think it's better immediately exec process.exit(1)

Reply:
could we should remove `fastify.close(exit)` and ` setTimeout(exit, 5000).unref()`  ?  it's  graceful exit looks never happend in uncaughtException. I think it's better immediately exec process.exit(1).